### PR TITLE
Upgrade to pac4j v1.8.2

### PR DESCRIPTION
**Pull request by leleuj**

### CODE REVIEW

File: play-pac4j-java/src/main/java/org/pac4j/play/CallbackController.java [hunk 1]
```diff
@@ -70,7 +67,7 @@ public Result callback() {
         try {
             credentials = client.getCredentials(context);
         } catch (final RequiresHttpAction e) {
-            return httpActionAdapter.handle(e.getCode(), context);
+            return (Result) config.getHttpActionAdapter().adapt(e.getCode(), context);
```
Comment (leleuj) [inline play-pac4j-java/src/main/java/org/pac4j/play/CallbackController.java]: @mmoayyed : notice the two use cases: 1) for an error, a specific Play result is generated (interactions with the `PlayWebContext` are not sufficient).

File: play-pac4j-java/src/main/java/org/pac4j/play/java/RequiresAuthenticationAction.java [hunk 2]
```diff
@@ -185,16 +181,15 @@ protected void saveRequestedUrl(final WebContext context, final List<Client> cur
     protected Result redirectToIdentityProvider(final PlayWebContext context, final List<Client> currentClients) {
         try {
             final IndirectClient currentClient = (IndirectClient) currentClients.get(0);
-            final RedirectAction action = currentClient.getRedirectAction(context, true);
-            logger.debug("redirectAction: {}", action);
-            return httpActionAdapter.handleRedirect(action);
+            currentClient.redirect(context, true);
+            return (Result) config.getHttpActionAdapter().adapt(context.getResponseStatus(), context);
```
Comment (leleuj) [inline play-pac4j-java/src/main/java/org/pac4j/play/java/RequiresAuthenticationAction.java]: @mmoayyed : 2) although a redirection has been computed on the web context, it must be turned into a specific Play result...

Comment (mmoayyed) [inline play-pac4j-java/src/main/java/org/pac4j/play/java/RequiresAuthenticationAction.java]: Looks good.

File: play-pac4j-java/src/main/java/org/pac4j/play/CallbackController.java (hunk 1 above)
Comment (mmoayyed) [inline play-pac4j-java/src/main/java/org/pac4j/play/CallbackController.java]: :+1:

**Pull request merged**